### ISSUE TITLE
mpv resource "docutils" cannot download because "Invalid certificate chain"

### ISSUE THREAD

**Issue by king7532** [Homebrew/homebrew-core#9008]
In the `mpv` formula, the URL for `docutils`:

>https://files.pythonhosted.org/packages/05/25/7b5484aca5d46915493f1fd4ecb63c38c333bd32aa9ad6e19da8d08895ae/docutils-0.13.1.tar.gz

 has the following problem "SSL certificate problem: Invalid certificate chain". The Safari web browser cannot open that URL and Chrome gives "NET::ERR_CERT_REVOKED"

```
==> Installing mpv 
/usr/bin/sandbox-exec -f /tmp/homebrew20170117-23715-9cswbo.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpv.rb --verbose
==> Downloading https://github.com/mpv-player/mpv/archive/v0.23.0.tar.gz
Already downloaded: /Users/king/Library/Caches/Homebrew/mpv-0.23.0.tar.gz
==> Verifying mpv-0.23.0.tar.gz checksum
tar xzf /Users/king/Library/Caches/Homebrew/mpv-0.23.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/05/25/7b5484aca5d46915493f1fd4ecb63c38c333bd32aa9ad6e19da8d08895ae/docutils-0.13.1.tar.gz
/usr/bin/curl --remote-time --location --user-agent Homebrew/1.1.7 (Macintosh; Intel macOS 10.12.2) curl/7.51.0 --fail https://files.pythonhosted.org/packages/05/25/7b5484aca5d46915493f1fd4ecb63c38c333bd32aa9ad6e19da8d08895ae/docutils-0.13.1.tar.gz -C 0 -o /Users/king/Library/Caches/Homebrew/mpv--docutils-0.13.1.tar.gz.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: Invalid certificate chain
More details here: https://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "mpv--docutils"
Download failed: https://files.pythonhosted.org/packages/05/25/7b5484aca5d46915493f1fd4ecb63c38c333bd32aa9ad6e19da8d08895ae/docutils-0.13.1.tar.gz
```

Is this resource required and could `mpv` be built without it?

**Comment by king7532** [comment]
@woodruffw what operating system are you using? Can you try on a mac? Thanks.

**Comment by king7532** [comment]
How can I get convince cURL to accept an invalid certificate chain? When I google, the answers say to open the link in Safari and then there is an option to always trust that host. But since Safari will not load that URL is there another way? Thanks

**Comment by king7532** [comment]
## I resolved this problem on my Mac by deleting the CRL and OCSP caches:

### OS X (through 10.11)

To delete both OCSP and CRL cache, in a terminal, enter the following command:
`sudo rm /var/db/crls/*cache?.db`

### OS X 10.12 Sierra

To delete both OCSP and CRL cache in OS X 10.12, open a terminal and run the following command:
`sqlite3 ~/Library/Keychains/*/ocspcache.sqlite3 'DELETE FROM ocsp;'`

@MikeMcQuaid: Could this added to the FAQ?

**Comment by woodruffw** [comment]
Can confirm that the download succeeds on a mac as well. Glad you resolved it.

Your local cert caches may have just been in a bad state, but I'm :+1: on adding this to the FAQ if more people report problems.

**Comment by king7532** [comment]
@woodruffw I have seen others report this problem while googling, which is why I asked if this could be added to the FAQ. Who is the gatekeeper of the Homebrew FAQ?

**Comment by woodruffw** [comment]
There's no real gatekeeper - it's just whatever common questions we receive.

Can you link me to some of those people reporting similar problems? One of our challenges is keeping the FAQ Homebrew-oriented - we can't cover every mistake/edge case/failure in OS X as a third party.

**Comment by king7532** [comment]
@woodruffw doing a Google search for [homebrew curl SSL certificate problem](https://www.google.com/search?client=safari&rls=en&q=homebrew+%22SSL+certificate+problem%22&ie=UTF-8&oe=UTF-8#q=homebrew+curl+SSL+certificate+problem) yields over 20,000 results, with various different ways to resolve the problem, like removing system supplied certificates 😬. I konw that I have run into this problem after updating macOS and would like to provide some clear instructions for others, include my future self.

**Comment by ilovezfs** [comment]
If you have another specific issue you yourself currently are seeing, please open a new ticket.

**Comment by woodruffw** [comment]
@king7532 I agree that there's inadequate documentation here, but this specific failure can have any number of causes that Homebrew has no control over:

* (Un)scheduled downtime
* Corporate proxies
* Corporate cert chains
* Skewed system time
* `~/.curlrc` (`brew doctor` complains about this one)
* Invalid/corrupt cert caches (your particular problem)

Any FAQ that tries to map a generic `curl` error to a specific solution is going to be met by someone who experiences the error for a different reason, and is irritated because our "official" solution doesn't work. OTOH, we might do well to add checks for some of these things to `brew doctor`.

**Comment by king7532** [comment]
Very well. I opened [rdar://30096489](http://openradar.appspot.com/30096489)
